README.md

```diff
@@ -34,22 +37,31 @@ See [instructions below](#usage-angular) to use a schematic to install dependenc
 
 Extend `@ni` in the [ESLint configuration](https://eslint.org/docs/user-guide/configuring/configuration-files#configuration-file-formats).
 
-```json
+```js
 {
-    "extends": "@ni"
+    extends: '@ni'
 }
 ```
 
 ### TypeScript
 
-Extend `@ni/eslint-config/typescript` in the ESLint configuration. Configure the `@typescript-eslint` plugin and the project's TypeScript configuration.
+Extend `@ni/eslint-config/typescript` in the default ESLint configuration. The default ESLint configuration is used during development (for example, in IDEs) and should only extend the TypeScript rules. The TypeScript rules requiring type checking should be run explicitly as an npm script to avoid performance issues during development.
 
-```json
+```js
+{
+    extends: '@ni/eslint-config/typescript'
+}
+```
+
+### TypeScript Rules Requiring Type Checking
+
+Extend `@ni/eslint-config/typescript-requiring-type-checking` in a separate ESLint configuration file (such as `.eslintrc-requiring-type-checking.js`) that includes a TypeScript configuration for `parserOptions.project`. Due to the longer execution time, the configuration requiring type checking should only be run from a `npm run lint` command explicitly invoked by developers and on CIs.
```

Comment:
Suggestion to distinguish the different lint commands a bit more clearly:

"...should not be included in the main `npm run lint` command. Prefer a separate `npm run lint-types` command explicitly invoked by developers and on CIs."

Reply:
Oh interesting. I was thinking there would be a single `npm run lint` command that runs both configurations. I expect during development most folks are relying on IDE feedback. I was thinking the cli would be run infrequently, ie when I'm ready to push and not continuously during development.

Reply:
How does this ensure that these rules are considered by IDE plugins. From the [documentation](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/TYPED_LINTING.md#performance):

"Most of our users are fine with this, as they think the power of type-aware static analysis is worth it. Additionally, most users primarily consume lint errors via IDE plugins which, through some caching magic, do not suffer the same penalties."

Reply:
Ah, good points. I was confused on multiple fronts but now that I understand how it works a little better, I agree that we don't need a separate `npm run lint-types` command.

If that "caching magic" is quick enough, then maybe we don't actually need a separate config file either. Let's discuss in our meeting today.

Reply:
We agreed in person to make our default recommendation to include type checking in IDEs and on builds. We can add a note mentioning splitting them as an option if performance is bad.